pilot/pkg/proxy/envoy/v2/internalgen.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"istio.io/istio/pilot/pkg/networking/util"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/golang/protobuf/ptypes/any"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	TypeURLConnect = "istio.io/connect"
```

Comment:
name bikeshed: these names represent events (because they are events), but XDS works in terms of resources. Is this the way they should be modeled? I could see a different approach like:

`istio.io/connectedInstances` - when a disconnect happens, we just remove it from the list - we don't send a disconnect event. Additionally, this makes more sense on initial connection to me. Right now you get 100s of "connect" events at startup, when they may have been connected for a long time

`istio.io/nackedResources` - resources remain in the collection until they are removed or are no longer NACKed

I am not sure what I mentioned is a good idea, just thinking out loud

Reply:
Sounds great, thanks for thinking out loud !

I'll change, it's a much better fit.

Reply:
We do need to propagate disconnect - so in future listeners can remove them from list. But that's fine - we can indicate the status in a field. Right now I'm sending the node meta of the node that is changing - to minimize conversion. I was planning to add a filed 'IP' with the peer IP, and I can add a 'status' as well. 

The purpose of this topic is to sync 'connected' status across Istiod and to other tools. The feed with the 100s of connect is working as intended ( but not yet fully implemented ). 

Eventually we can define a dedicated proto, or use EndpointSlice.

Reply:
For disconnect, can we just remove it from the response? Like when an endpoint is removed